### PR TITLE
Updated Service discovery (Consul service discovery) for Choreo Connect

### PR DESCRIPTION
**Pull request by rumeshmadhusanka**
Screenshot from 2021-04-16 21-25-17
Screenshot from 2021-04-16 21-25-37
Screenshot from 2021-04-16 21-25-45
Screenshot from 2021-04-16 21-26-00
Screenshot from 2021-04-16 21-26-08
Screenshot from 2021-04-16 21-27-39
Screenshot from 2021-04-16 21-27-56
Screenshot from 2021-04-16 21-28-19
Screenshot from 2021-04-16 21-28-47
Screenshot from 2021-04-16 21-28-57
Screenshot from 2021-04-16 21-29-14

### CODE REVIEW

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Configure Choreo Connect with Consul service mesh deployed on Kubernetes
+Example:
+
+![service discovery](../../../../assets/img/deploy/consul-reference-k8s.png)
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md:4]:
```suggestion
![service discovery]({{base_path}}/assets/img/deploy/consul-reference-k8s.png)
```
Always use base_path for links.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Configure Choreo Connect with Consul service mesh deployed on Kubernetes
+Example:
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md:2]: Please add some kind of explanation or overview that covers the following:

- What are we trying to achieve with this documentation.
- Description of what Consul service mesh is.
- Why someone would want to connect Choreo Connect with Consul.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+![service discovery](../../../../assets/img/deploy/consul-reference-k8s.png)
+
+## Step 1 - Deploy Consul on K8s
+Deploy Consul using the [official Helm Chart](https://helm.releases.hashicorp.com/).<br>
+For more instructions, see the [Consul official documentation](https://www.consul.io/docs/k8s).
+## Step 2 - Modify Choreo Connect k8s artifacts 
+Add the `'consul.hashicorp.com/connect-inject': 'true'` annotation to Adapter's deployment spec
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md:10]:
```suggestion
Add the `'consul.hashicorp.com/connect-inject': 'true'` annotation to the Adapter's deployment specification.
```
Shall we add which file exactly we are meant to modify here?

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    matchLabels:
+      app: choreo-connect-adapter
+  template:
+    metadata:
+      labels:
+        app: choreo-connect-adapter
+      annotations:
+        'consul.hashicorp.com/connect-inject': 'true'
+```
+In the config.toml config map, go to the \[adapter.consul] section and update it's `url` with k8s node ip.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md:28]:
```suggestion
In the config.toml config file, go to the \[adapter.consul] section and update the `url` with the IP address of the Kubernetes node.
```

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-discovery.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Configure Choreo Connect with Consul Discovery (without service mesh)
+
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-discovery.md:2]: Please add an overview similar to the above covering the following.

- What are we trying to achieve with this documentation.
- Description of what Consul Discovery is.
- Why someone would want to connect Choreo Connect with Consul Discovery.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-discovery.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  caCertFile = "/home/wso2/security/truststore/consul/consul-agent-ca.pem"
+  certFile = "/home/wso2/security/truststore/consul/local-dc-client-consul-0.pem"
+  keyFile = "/home/wso2/security/truststore/consul/local-dc-client-consul-0-key.pem"
+```
+Change the `url` to the url of the Consul agent you want to connect to.<br>
+Change the `aclToken` to the token you generated from the previous step.<br>
+`serviceMeshEnabled` should be set to `false`.<br>
+You can leave the `mgwServiceName` empty as the property is not needed.<br>
+Add the `caCertFile`, `certFile`, and `keyFile` you generated from the previous step.<br>
+Then start Choreo Connect.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-discovery.md:27]: Let's convert these items into a numbered list.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
+# Configure Choreo Connect with Consul service mesh
+Example:
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md:2]: Same comment as above. Please add an appropriate overview.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  certFile = "/home/wso2/security/truststore/consul/local-dc-client-consul-0.pem"
+  keyFile = "/home/wso2/security/truststore/consul/local-dc-client-consul-0-key.pem"
+```
+
+Change the `url` to the url of the Consul agent you want to connect to.<br>
+Change the `aclToken` to the token you generated from the previous step.<br>
+Make sure the property `serviceMeshEnabled` is set to `true`.<br>
+`mgwServiceName` should be the name you registered Choreo Connect as Consul service.<br>
+Add the `caCertFile`, `certFile`, and `keyFile` you generated from the previous step.<br>
+Then start Choreo Connect.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md:57]: Convert to numbered list.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+## Step 4 - Deploy the API
+You can use either WSO2 API Manager or APICTL to deploy APIs as described in [how to define the endpoints](../service-discovery/#how-to-define-the-endpoints) guide.
```
Comment (samgnaniah) [inline en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md:60]: Use base_path for these links as well.

File: en/mkdocs.yml [hunk 10]
```diff
@@ -251,7 +251,11 @@ nav:
                 - HTTP 2.0: deploy-and-publish/deploy-on-gateway/choreo-connect/http-2.0.md
                 - Rate Limiting:
                     - Enabling distributed throttling: deploy-and-publish/deploy-on-gateway/choreo-connect/rate-limiting/distributed-throttling.md
-                - Service Discovery: deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery.md
+                - Service Discovery:
+                    - Configure Choreo Connect with Consul: deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/service-discovery.md
+                    - Service Discovery:  deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-discovery.md
```
Comment (samgnaniah) [inline en/mkdocs.yml:256]: Shall we make this the first topic? We can rename this to Service Discovery Overview.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md (hunk 3 above)
Comment (rumeshmadhusanka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-k8s.md]: fixed 364f6cb

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md (hunk 9 above)
Comment (rumeshmadhusanka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md:60]: fixed d0bdd2e

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md (hunk 8 above)
Comment (rumeshmadhusanka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md]: fixed 3acd332

File: en/mkdocs.yml (hunk 10 above)
Comment (rumeshmadhusanka) [reply en/mkdocs.yml:256]: I changed **Configure Choreo Connect with Consul** --> **Service Discovery Overview**
Guess that's what you meant
5378648

Comment (samgnaniah) [reply en/mkdocs.yml:256]: I meant change Service Discovery (service-discovery.md) to Service Discovery Overview.

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md (hunk 7 above)
Comment (rumeshmadhusanka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-mesh.md:11]: Screenshot from 2021-04-22 08-55-01

File: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-discovery.md (hunk 5 above)
Comment (rumeshmadhusanka) [reply en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/service-discovery/choreo-connect-and-service-discovery.md:2]: Screenshot from 2021-04-22 07-52-14

**Review by samgnaniah — APPROVED** [review]

**Pull request merged**